http-auth/routes/routes.go

```diff
@@ -0,0 +1,98 @@
+package routes
+
+import (
+	"fmt"
+	"html"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"golang.org/x/time/rate"
+)
+
+const HTMLHeader = "<!DOCTYPE html><html>"
+
+func IsTestRun() bool {
+	return os.Getenv("EXECUTION_ENVIRONMENT") == "test"
+}
+
+func IndexHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "GET" {
+		w.Header().Set("Content-Type", "text/html")
+		b := "<em>Hello World</em>"
+
+		queryParams := r.URL.Query()
+
+		if len(queryParams) > 0 {
+			b = "<h1>Query Params</h1><ul>"
+			for key, value := range queryParams {
+				b += fmt.Sprintf("<li>%s: %s</li>", html.EscapeString(key), html.EscapeString(strings.Join(value, ", ")))
+			}
+			b += "</ul>"
+		}
+
+		body := fmt.Sprintf("%s%s", HTMLHeader, b)
+		w.Write([]byte(body))
+	}
+
+	if r.Method == "POST" {
+		w.Header().Set("Content-Type", "text/html")
+		returnMessage := ""
+
+		b, err := io.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte("Error reading request body"))
```

Comment:
What do you expect to be returned to the user in this case? Do you have a test which shows what's returned to the user in this case? Can you write one? I don't think it's returning exactly what you expect!

Reply:
Not sure what you mean here. I have written a test and if `io.ReadAll` returns an error then the response body is what I expected it to be, i.e `Error reading request body`

I have added a test that checks this, but not 100% sure of the approach. I have modified the handler to accept a reader so I can provide a mockReader which throws an error